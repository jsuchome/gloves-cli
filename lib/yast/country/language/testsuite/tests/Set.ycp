/**
 * tests for Language::Set
 * Author:	jsuchome@suse.cz
 * $Id: Set.ycp 32769 2006-09-08 09:29:15Z jsuchome $
 */
{

    // testedfiles: Language.ycp

    include "testsuite.ycp";

    map READ = $[
	"sysconfig" : $[
	    "language" : $[
		"RC_LANG"	: "en_US.UTF-8",
	    ],
	],
	"target" : $[
	    "bash_output"	: $[],
	    "size"		: 1,
	    "yast2"		: $[],
	],
    ];

    TESTSUITE_INIT([READ,$[],READ], nil);

    import "Language";

    map R	= $[
	"target" : $[
	    "dir" : [
		"language_de_DE.ycp"
	    ],
	    "yast2" : $[
		"de_DE" : [
		    "Deutsch",
		    "Deutsch",
		    ".UTF-8",
		    "@euro",
		    "German"
		],
	    ],
	],
    ];

    TEST(``(Language::GetLocaleString (Language::language)), [R,$[],$[]], nil);

    map EX	= $[
	"target" : $[
	    "bash_output"       : $[],
	],
    ];

    TEST(``(Language::Set ("de_DE")), [$[],$[],EX], nil);

    TEST(``(Language::GetLocaleString (Language::language)), [], nil);

    map expert	= $[
	"use_utf8"	: false
    ];
    TEST(``(Language::SetExpertValues (expert)), [], nil);
    TEST(``(Language::GetLocaleString (Language::language)), [], nil);
}
