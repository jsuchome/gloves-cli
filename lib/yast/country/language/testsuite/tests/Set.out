Read	.target.size "/etc/sysconfig/language" 1
Read	.target.dir "/usr/share/YaST2/data/languages" ["language_de_DE.ycp"]
Read	.target.yast2 "languages/language_de_DE.ycp" $["de_DE":["Deutsch", "Deutsch", ".UTF-8", "@euro", "German"]]
Return	en_US.UTF-8
Execute	.target.bash_output "/usr/bin/locale -a" $[]
Return	nil
Return	de_DE.UTF-8
Return	nil
Return	de_DE@euro
